app/src/main/java/com/example/chickenmasala/ui/screen/GuessTheGameFragment.kt

```diff
@@ -0,0 +1,178 @@
+import com.example.chickenmasala.data.utils.RecipeParser
+import com.example.chickenmasala.databinding.FragmentGuessTheMealBinding
+import com.example.chickenmasala.util.Constants
+import kotlinx.coroutines.*
+
+class GuessTheGameFragment(private val gameName: GuessGamesName) :
+    BaseFragment<FragmentGuessTheMealBinding>() {
+    override val LOG_TAG: String = "GuessTheMealFragment"
+    override val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> FragmentGuessTheMealBinding =
+        FragmentGuessTheMealBinding::inflate
+    private lateinit var guessTheMeal: QuestionGames
+    private var selectedChoiceId: Int = 0
+    private var yellowColor: Int? = null
+    private var grayColor: Int? = null
+    private var greenColor: Int? = null
+    private var redColor: Int? = null
+    private var multiChoices: List<TextView>? = null
+    override fun setup() {
+        val recipeParser = RecipeParser()
+        val dataSource =
+            CsvDataSource(requireContext(), Constants.RECIPES_CSV_FILE_NAME, recipeParser)
+        guessTheMeal = GuessGamesInteractor(dataSource).guessGames(gameName)
+        multiChoices =
+            listOf(
+                binding.answerOneText,
+                binding.answerTwoText,
+                binding.answerThreeText,
+                binding.answerFourText
+            )
+        showGame()
+    }
+
+    private fun showGame() {
+        prepareColors()
+        prepareMultiChoices()
+        prepareQuestion()
+        setButtonStatus(binding.submitButton, false)
+        prepareNextQuestion(clickChoicesState = true)
+    }
+
+    override fun addCallBacks() {
+        setColorOfSelectedChoice()
+        submitAnswer()
+    }
+
+    private fun prepareColors() {
+        yellowColor = ContextCompat.getColor(requireContext(), R.color.yellow_600)
+        grayColor = ContextCompat.getColor(requireContext(), R.color.gray_200)
+        greenColor = ContextCompat.getColor(requireContext(), R.color.green_100)
+        redColor = ContextCompat.getColor(requireContext(), R.color.red_100)
+        setUpChoicesColor()
+    }
+
+    private fun setUpChoicesColor() {
+        multiChoices?.forEach { choice ->
+            colorizeTheView(choice, grayColor!!)
+        }
+    }
+
+    private fun prepareMultiChoices() {
+        val listOfChoices = getRandomArrangeOfChoices()
+        binding.apply {
+            answerOneText.text = listOfChoices[0]
+            answerTwoText.text = listOfChoices[1]
+            answerThreeText.text = listOfChoices[2]
+            answerFourText.text = listOfChoices[3]
+        }
+    }
+
+    private fun getRandomArrangeOfChoices(): List<String> {
+        return listOf(
+            guessTheMeal.correctAnswer,
+            guessTheMeal.wrongAnswer1,
+            guessTheMeal.wrongAnswer2,
+            guessTheMeal.wrongAnswer3
+        )
+            .shuffled()
+    }
+
+    private fun prepareQuestion() {
+        Glide.with(this)
```

Comment:
use binding.apply and replace this to mealImageView

Reply:
should i use scope function for one view ?